pastis/matrix_generation/matrix_from_efields.py

```diff
@@ -88,14 +90,19 @@ def pastis_matrix_from_efields(electric_fields, efield_ref, direct_norm, dh_mask
     return matrix_pastis
 
 
-def calculate_semi_analytic_pastis_from_efields(efields, efield_ref, direct_norm, dh_mask):
+def calculate_semi_analytic_pastis_from_efields(instrument, efields, efield_ref, direct_norm, dh_mask):
 
     # Create empty matrix
     nb_modes = efields.shape[0]
     matrix_pastis_half = np.zeros([nb_modes, nb_modes])
 
     for pair in util.segment_pairs_non_repeating(nb_modes):
-        intensity_im = np.real((efields[pair[0]].electric_field - efield_ref) * np.conj(efields[pair[1]].electric_field - efield_ref))
+        if instrument == 'RST':
+            intensity_im = np.real(
+                (efields[pair[0]].wavefront - efield_ref) * np.conj(efields[pair[1]].wavefront - efield_ref))
+        elif instrument == 'LUVOIR':
+            intensity_im = np.real(
+                (efields[pair[0]].electric_field - efield_ref) * np.conj(efields[pair[1]].electric_field - efield_ref))
```

Comment:
We should try to find a more elegant solution for this. My suggestion would be to make the input parameter `efields` already carry the actual E-fields, rather than some third-party object instance. For this too, let's talk offline.